src/plots/polar/layout_attributes.js

```diff
@@ -0,0 +1,327 @@
+    // to draw donut-like charts
+    // e.g. https://github.com/matplotlib/matplotlib/issues/4217
+    //
+    // maybe something like 'span' or 'hole' (like pie, but pie set it in data coords?)
+    // span: {},
+    // hole: 1
+
+    // maybe should add a boolean to enable square grid lines
+    // and square axis lines
+    // (most common in radar-like charts)
+    // e.g. squareline/squaregrid or showline/showgrid: 'square' (on-top of true)
+
+    editType: 'calc'
+};
+
+extendFlat(
+    radialAxisAttrs,
+
+    // N.B. the radialaxis grid lines are circular,
+    // but radialaxis lines are straight from circle center to outer bound
+    axisLineGridAttr,
+    axisTickAttr
+);
+
+var angularAxisAttrs = {
+    visible: extendFlat({}, axesAttrs.visible, {dflt: true}),
+    type: {
+        valType: 'enumerated',
+        // 'linear' should maybe be called 'angle' or 'angular' here
+        // to make clear that axis here is periodic and more tightly match
+        // `thetaunit`?
+        //
+        // no 'log' for now
+        values: ['-', 'linear', 'date', 'category'],
+        dflt: '-',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the angular axis type.',
+            'If *linear*, set `thetaunit` to determine the unit in which axis value are shown.',
+            'If *date*, set `period` to determine the wrap around period.',
+            'If *category, set `period` to determine the number of integer coordinates around polar axis.'
+        ].join(' ')
+    },
+
+    categoryorder: axesAttrs.categoryorder,
+    categoryarray: axesAttrs.categoryarray,
+
+    thetaunit: {
+        valType: 'enumerated',
+        values: ['radians', 'degrees'],
+        dflt: 'degrees',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the format unit of the formatted *theta* values.',
+            'Has an effect only when `angularaxis.type` is *linear*.'
+        ].join(' ')
+    },
+
+    period: {
+        valType: 'any',
+        editType: 'calc',
+        role: 'info',
+        description: ''
+
+        // 360 / 2*pi for linear (might not need to set it)
+        // and to full range for other types
+
+        // 'period' is the angular equivalent to 'range'
+
+        // similar to dtick, one way to achieve e.g.:
+        // - period that equals the timeseries length
+        //  http://flowingdata.com/2017/01/24/one-dataset-visualized-25-ways/18-polar-coordinates/
+        // - and 1-year periods (focusing on seasonal change0
+        //  http://otexts.org/fpp2/seasonal-plots.html
+        //  https://blogs.scientificamerican.com/sa-visual/why-are-so-many-babies-born-around-8-00-a-m/
+        //  http://www.seasonaladjustment.com/2012/09/05/clock-plot-visualising-seasonality-using-r-and-ggplot2-part-3/
+        //  https://i.pinimg.com/736x/49/b9/72/49b972ccb3206a1a6d6f870dac543280.jpg
+        //  https://www.climate-lab-book.ac.uk/spirals/
+    },
```

Comment:
Some screenshot of date angular axes:

![image](https://user-images.githubusercontent.com/6675409/33892182-285e02ec-df26-11e7-92c4-71f864820b34.png)

![image](https://user-images.githubusercontent.com/6675409/33892200-32ec6104-df26-11e7-9215-d702616a1f29.png)

![image](https://user-images.githubusercontent.com/6675409/33892242-49039ef8-df26-11e7-875f-96618c712e43.png)


![image](https://user-images.githubusercontent.com/6675409/33892282-6bc194f4-df26-11e7-83aa-e365db28a934.png)

Reply:
Great feature !! Can you please include the code (layout config attributes) for the above example screenshots?